Assets/Scripts/EOSAchievementManager.cs

```diff
@@ -275,22 +184,61 @@ private static AchievementsInterface GetEOSAchievementInterface()
         /// Invoked when the request from the server has completed - whether
         /// successful or not.
         /// </param>
-        private static void QueryAchievements(ProductUserId productUserId, OnQueryDefinitionsCompleteCallback callback)
+        private Task<List<DefinitionV2>> QueryAchievementsAsync(ProductUserId productUserId)
         {
             var options = new QueryDefinitionsOptions
             {
                 LocalUserId = productUserId
             };
 
+            TaskCompletionSource<List<DefinitionV2>> tcs = new();
+
             GetEOSAchievementInterface().QueryDefinitions(ref options, null, (ref OnQueryDefinitionsCompleteCallbackInfo data) =>
             {
                 if (data.ResultCode != Result.Success)
                 {
                     Log($"Unable to query achievement definitions. Result code: {data.ResultCode}");
                 }
 
-                callback?.Invoke(ref data);
+                tcs.SetResult(GetCachedAchievements());
             });
+
+            return tcs.Task;
+        }
+
+        /// <summary>
+        /// Retrieves the list of achievement definitions for the game that have
+        /// been cached locally.
+        /// </summary>
+        /// <returns>A list of achievement definitions.</returns>
+        private List<DefinitionV2> GetCachedAchievements()
+        {
+            uint achievementDefinitionCount = GetAchievementsCount();
+            var options = new CopyAchievementDefinitionV2ByIndexOptions
+            {
+                AchievementIndex = 0
+            };
+
+            List<DefinitionV2> achievements = new();
+
+            for (uint i = 0; i < achievementDefinitionCount; ++i)
+            {
+                options.AchievementIndex = i;
+                GetEOSAchievementInterface().CopyAchievementDefinitionV2ByIndex(ref options, out DefinitionV2? definition);
+
+                // Move on if the definition is empty
+                if (!definition.HasValue)
+                    continue;
+
+                achievements.Add(definition.Value);
+
+                UnityEngine.Debug.LogFormat("Achievements (CacheAchievementDef): Id={0}, LockedDisplayName={1}", definition.Value.AchievementId, definition.Value.LockedDisplayName);
```

Comment:
Should this use the `print` functions so that it can be conditionally excluded from prod builds?

Reply:
Good idea. I've updated it to do so.